internal/servers/controller/handlers/sessions/session_service.go

```diff
@@ -79,35 +79,55 @@ func (s Service) ListSessions(ctx context.Context, req *pbs.ListSessionsRequest)
 	if err := validateListRequest(req); err != nil {
 		return nil, err
 	}
-	authResults := s.authResult(ctx, req.GetScopeId(), action.List)
-	if authResults.Error != nil {
-		return nil, authResults.Error
+	sl, err := s.listSessions(ctx, action.List, req.GetScopeId(), req.GetRecursive())
+	if err != nil {
+		return nil, err
 	}
+	return &pbs.ListSessionsResponse{Items: sl}, nil
+}
 
-	scopeIds, scopeInfoMap, err := scopeids.GetScopeIds(
-		ctx, s.iamRepoFn, authResults, req.GetScopeId(), req.GetRecursive())
+// ListSelfSessions implements the interface pbs.SessionServiceServer.
+func (s Service) ListSelfSessions(ctx context.Context, req *pbs.ListSelfSessionsRequest) (*pbs.ListSelfSessionsResponse, error) {
+	if err := validateListSelfRequest(req); err != nil {
+		return nil, err
+	}
+	sl, err := s.listSessions(ctx, action.ListSelf, req.GetScopeId(), req.GetRecursive())
 	if err != nil {
 		return nil, err
 	}
+	return &pbs.ListSelfSessionsResponse{Items: sl}, nil
+}
 
-	seslist, err := s.listFromRepo(ctx, scopeIds)
+func (s Service) listSessions(ctx context.Context, a action.Type, scopeId string, recursive bool) ([]*pb.Session, error) {
+	authResults := s.authResult(ctx, scopeId, a)
+	if authResults.Error != nil {
+		return nil, authResults.Error
+	}
+	scopeIds, scopeInfoMap, err := scopeids.GetScopeIds(ctx, s.iamRepoFn, authResults, scopeId, recursive)
+	if err != nil {
+		return nil, err
+	}
+	opts := []session.Option{session.WithScopeIds(scopeIds)}
+	if a == action.ListSelf {
+		opts = append(opts, session.WithUserId(authResults.UserId))
+	}
+	seslist, err := s.listFromRepo(ctx, opts...)
 	if err != nil {
 		return nil, err
 	}
-
 	finalItems := make([]*pb.Session, 0, len(seslist))
 	res := &perms.Resource{
-		Type: resource.Session,
+		ScopeId: authResults.Scope.Id,
+		Type:    resource.Session,
 	}
 	for _, item := range seslist {
 		item.Scope = scopeInfoMap[item.GetScopeId()]
-		res.ScopeId = item.Scope.Id
 		item.AuthorizedActions = authResults.FetchActionSetForId(ctx, item.Id, IdActions, auth.WithResource(res)).Strings()
```

Comment:
Not a comment on this line but I can't comment above -- CollectionActions should have ListSelf added to it, at the top of the file.

Reply:
Oh also -- we should add list:self to default role permissions for authenticated users for global/org scopes. We don't need to upgrade existing roles, that's up to users to decide if they want to add it.

Reply:
Added.  Thanks!